ec2utils/ec2publishimg/ec2publishimg

```diff
@@ -0,0 +1,225 @@
+    dest='dryRun',
+    help=help_msg
+)
+argparse.add_argument(
+    '-f', '--file',
+    default=os.path.expanduser('~') + os.sep + '.ec2utils.conf',
+    dest='configFilePath',
+    help='Path to configuration file, default ~/.ec2utils.conf (Optional)',
+    metavar='CONFIG_FILE'
+)
+# Note, one of the arguments in the group is required if --version is
+# not specified. However setting this behavior through the parser
+# also requiers one argument to be specified even if --version is specified
+# This parser behavior is true even if --version and the group are part of the
+# same subgroup
+publish_image_condition_group = argparse.add_mutually_exclusive_group()
+publish_image_condition_group.add_argument(
+    '--image-id',
+    dest='pubImgID',
+    help='The AMI ID of the image to be published (Optional)',
+    metavar='AMI_ID'
+)
+publish_image_condition_group.add_argument(
+    '--image-name',
+    dest='pubImgName',
+    help='The image name of the image to be published (Optional)',
+    metavar='IMAGE_NAME'
+)
+help_msg = 'An image name fragment to match the image name of the image to be '
+help_msg += 'published (Optional)'
+publish_image_condition_group.add_argument(
+    '--image-name-frag',
+    dest='pubImgNameFrag',
+    help=help_msg,
+    metavar='IMAGE_NAME_FRAGMENT'
+)
+help_msg = 'A regular expression to match the image name of the image to be '
+help_msg += 'published (Optional)'
+publish_image_condition_group.add_argument(
+    '--image-name-match',
+    dest='pubImgNameMatch',
+    help=help_msg,
+    metavar='REGEX'
+)
+help_msg = 'Comma separated list of regions for publishing, all integrated '
+help_msg += 'region sif not given (Optional)'
+argparse.add_argument(
+    '-r', '--regions',
+    dest='regions',
+    help=help_msg,
+    metavar='EC2_REGIONS'
+)
+argparse.add_argument(
+    '-s', '--secret-key',
+    dest='secretKey',
+    help='AWS secret access key (Optional)',
+    metavar='AWS_SECRET_KEY'
+)
+argparse.add_argument(
+    '--share-with',
+    default='all',
+    dest='share',
+    help='all, none, or a AWS account number to share the image(s) with',
+    metavar='SHARE'
+)
+argparse.add_argument(
+    '--verbose',
+    action='store_true',
+    default=False,
+    dest='verbose',
+    help='Enable on verbose output'
+)
+argparse.add_argument(
+    '--version',
+    action='store_true',
+    default=False,
+    dest='version',
+    help='Program version'
+)
+
+args = argparse.parse_args()
+
+if args.version:
+    import ec2utils
+    version_file_name = 'publish_VERSION'
+    base_path = os.path.dirname(ec2utils.__file__)
+    version = open(base_path + os.sep + version_file_name, 'r').read()
+    print version
+    sys.exit(0)
+
+# Explicit check required to to the group issue, see comment above
+if (
+        not args.pubImgID
+        and not args.pubImgName
+        and not args.pubImgNameFrag
+        and not args.pubImgNameMatch):
+    error_msg = 'ec2publishimg: error: one of the arguments '
```

Comment:
The prefix is there to stay in line with the error style argparse would produce for a missing argument. However, as noted in the comment the group cannot be handled by argparse, thus the goal was to make the error appear in the same style even though it is generated by this code rather than argparse.